core/iwasm/common/wasm_loader_common.c

```diff
@@ -104,7 +104,9 @@ bool
 is_valid_func_type(const WASMFuncType *func_type)
 {
     unsigned i;
-    for (i = 0; i < func_type->param_count + func_type->result_count; i++) {
+    for (i = 0;
+         i < (unsigned)(func_type->param_count + func_type->result_count);
```

Comment:
what problem does this solve?

Reply:
the type of `func_type->param_count` and `func_type->result_count` is uint16, when comparing to `unsigned i`, `func_type->param_count + func_type->result_count` is converted from uint16 into signed int, and the compiler reported warning of comparison between signed and unsigned.